### ISSUE TITLE
About sorting a many:many relation result set

### ISSUE THREAD

**Issue by Fastfuel** [ikkez/f3-cortex#77]
Hi Ikkez,
It's more a question rather than an issue (tell me if SO is better place for questions). I'm looking for the best way to implement a sorting solution to display photos in Portfolio. I have configure a _m:m_ relation between portfolios and photos in their respective models, with a custom pivot table name and it's working fine.
Does Cortex handle the sorting on an extra field on the pivot table (via filters) or must I do it via the orderBy CortexCollection method after having fetch the photo set ? Any other way to do it ? Thanks !

**Comment by ikkez** [comment]
Hi there. I'm not quite sure.. so you want to sort by the extra field in the pivot table?! I think that field isnt really there after fetching the resultsl. You can of course sort the relation via a filter, yes.. but if that field in on the pivot table, you either need to make a new model from the pivot table itself and use that as aggregation point for find() or help me to hack a new filter into the point where the pivot table is queried..which should be right here: https://github.com/ikkez/f3-cortex/blob/master/lib/db/cortex.php#L1788 and/or maybe here: https://github.com/ikkez/f3-cortex/blob/master/lib/db/cortex.php#L1833
basically we need to fill the `null` in `$rel->find($filter,null,$this->_ttl);` which an options array

**Comment by Fastfuel** [comment]
Hi Ikkez,
Thank you for your answer.

> you either need to make a new model from the pivot table itself and use that as aggregation point for find()

Doing so i'll retrieve a sorted cortexCollection of relations and I'm loosing the easy way provided by Cortex to deal with datas.

> or help me hack… with an options array

I'ld love to but my developper skills are far from that :-) I was just thinking that maybe the pivot table, when custom defined and having a corresponding model, can be automatically used to refine m:m relations. For example adding an "order" field in the pivot table and in the $fieldConf of it's model can be used when Cortex retrieves or save the related models... 

Thanks  anyway.

**Comment by ikkez** [comment]
Well that actually is the case.. you can use the `->filter()` method to sort relations by a defined field or something.. I can however understand why adding an order/sorting field to the pivot table could be useful.. I'm thinking about it.. in the meanwhile however, you can just sort the data while saving the reation, and it "should" be received in the same order... meaning `$portfolio->photos = [54,122,1,5];`.

**Comment by Fastfuel** [comment]
Good to know that it could be in the next release. I'll try this last trick.
Thanks !
